lib/app.js

```diff
@@ -22,78 +17,47 @@ const otpConfig = require(YAML_CONFIG)
 const jsConfig = require(JS_CONFIG).configure(otpConfig)
 
 const {
+  defaultMobileTitle = null,
   ItineraryBody,
-  getItineraryFooter,
-  getContainerSlot,
-  getMapContainerSlot,
-  getMainPanel,
+  ItineraryFooter,
   LegIcon,
-  ModeIcon
+  MainControls,
+  MainPanel,
+  MapWindows,
+  MobileResultsScreen,
+  MobileSearchScreen,
+  ModeIcon,
+  TermsOfService,
+  TermsOfStorage
 } = jsConfig
 
-// Helper method to get component if the getMethod param is a valid function
-const getComponentIfExists = (getMethod) => {
-  return typeof getMethod === 'function'
-    ? getMethod()
-    : null
-}
-
-const mainPanel = getComponentIfExists(getMainPanel)
-// Main panel and Leg/Mode icons are required.
-if (!mainPanel || !LegIcon || !ModeIcon || !ItineraryBody) {
-  throw new Error('ItineraryBody, getMainPanel, LegIcon, and ModeIcon must be defined in config.js')
+// Check that the required components are defined in the configuration.
+if (
+  !ItineraryBody ||
+  !LegIcon ||
+  !MainPanel ||
+  !MobileResultsScreen ||
+  !MobileSearchScreen ||
+  !ModeIcon
+) {
+  throw new Error('ItineraryBody, LegIcon, MainPanel, MobileResultsScreen, MobileSearchScreen, and ModeIcon must be defined in config.js')
 }
 
 const components = {
+  defaultMobileTitle,
   ItineraryBody,
-  itineraryFooter: getComponentIfExists(getItineraryFooter),
+  ItineraryFooter,
   LegIcon,
-  ModeIcon
+  MainControls,
+  MainPanel,
+  MapWindows,
+  MobileResultsScreen,
+  MobileSearchScreen,
+  ModeIcon,
+  TermsOfService,
+  TermsOfStorage
 }
 
-export default class TrimetWebapp extends Component {
-  render () {
-    const {branding} = otpConfig
-    /** desktop view **/
-    const desktopView = (
-      <div className='otp'>
-        <DesktopNav />
-        <Grid>
-          <Row className='main-row'>
-            <Col sm={6} md={4} className='sidebar'>
-              {/* <main> is needed for accessibility checks. */}
-              <main>
-                {mainPanel}
-              </main>
-            </Col>
-            {getComponentIfExists(getContainerSlot)}
-            <Col sm={6} md={8} className='map-container'>
-              {getComponentIfExists(getMapContainerSlot)}
-              <Map />
-            </Col>
-          </Row>
-        </Grid>
-      </div>
-    )
+const TrimetWebapp = () => <ResponsiveWebapp components={components} />
```

Comment:
Should we just write `<ResponsiveWebapp components={jsConfig} />`?

Reply:
I'd be fine with that if it works.

Reply:
Updated in https://github.com/ibi-group/trimet-mod-otp/pull/292/commits/5bc212d92aba21da438632e14aaa74dd78e7654f.